src/Compilers/CSharp/Portable/Binder/Binder_Statements.cs

```diff
@@ -3537,18 +3538,26 @@ private BoundCatchBlock BindCatchBlock(CatchClauseSyntax node, ArrayBuilder<Boun
             return new BoundCatchBlock(node, locals, exceptionSource, type, boundFilter, block, hasError);
         }
 
-        private BoundExpression BindCatchFilter(CatchFilterClauseSyntax filter, DiagnosticBag diagnostics)
+        private BoundExpression BindCatchFilter(CatchFilterClauseSyntax filter, bool hasSingleCatch, DiagnosticBag diagnostics)
         {
             BoundExpression boundFilter = this.BindBooleanExpression(filter.FilterExpression, diagnostics);
             if (boundFilter.ConstantValue != ConstantValue.NotAvailable)
             {
-                Error(diagnostics, ErrorCode.WRN_FilterIsConstant, filter.FilterExpression);
+                // Depending on whether the filter constant is true or false, and whether there are other catch clauses,
+                // we suggest different actions
+                var errorCode = boundFilter.ConstantValue.BooleanValue
+                    ? ErrorCode.WRN_FilterIsConstantTrue
+                    : hasSingleCatch
```

Comment:
Rather than passing around hasSingleCatch, I think it would be cleaner to compute it here (from `filter.Parent.Parent`).